container/generate-dockerfile.py

```diff
@@ -0,0 +1,167 @@
+    ("io.k8s.display-name", "OpenSCAP"),
+    ("io.k8s.description", "OpenSCAP is an auditing tool that utilizes the Extensible Configuration Checklist Description Format (XCCDF). XCCDF is a standard way of expressing checklist content and defines security checklists."),
+    ("io.openshift.tags", "security openscap scan"),
+    ("install", "docker run --rm --privileged -v /:/host/ IMAGE sh /root/install.sh"),
+    ("run", "docker run -it --rm -v /:/host/ IMAGE sh /root/run.sh")
+]
+packages = [
+    "bzip2",
+    "wget"
+]
+files = [
+    ("install.sh", "/root"),
+    ("run.sh", "/root"),
+    ("openscap", "/root"),
+    ("config.ini", "/root")
+]
+env_variables = [
+    ("container", "docker")
+]
+install_commands = {
+    "fedora": "dnf",
+    "rhel": "yum"
+}
+builddep_packages = {
+    "fedora" : "'dnf-command(builddep)'",
+    "rhel" : "yum-utils"
+}
+builddep_commands = {
+    "fedora": "dnf -y builddep",
+    "rhel": "yum-builddep -y"
+}
+download_cve_feeds_command = [
+    "wget --no-verbose -P /var/lib/oscapd/cve_feeds/ "
+    "https://www.redhat.com/security/data/oval/com.redhat.rhsa-RHEL{5,6,7}.xml.bz2",
+    "bzip2 -dk /var/lib/oscapd/cve_feeds/com.redhat.rhsa-RHEL{5,6,7}.xml.bz2",
+    "ln -s /var/lib/oscapd/cve_feeds/ /var/tmp/image-scanner"
+]
+openscap_build_command = [
+    "git clone -b maint-1.2 https://github.com/OpenSCAP/openscap.git",
+    "pushd /openscap",
+    "./autogen.sh",
+    "./configure --enable-sce --prefix=/usr",
+    "make -j 4 install",
+    "popd"
+]
+ssg_build_command = [
+    "git clone https://github.com/OpenSCAP/scap-security-guide.git",
+    "pushd /scap-security-guide/build",
+    "cmake -DCMAKE_INSTALL_DATADIR=/usr/share ..",
+    "make -j 4 install",
+    "popd"
+]
+daemon_build_command = [
+    "git clone https://github.com/OpenSCAP/openscap-daemon.git",
+    "pushd /openscap-daemon",
+    "python setup.py install",
+    "popd"
+]
+delim = " && \\\n    "
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Builds an image with OpenSCAP Daemon")
+    parser.add_argument("--base", type=str, default="fedora", help="Base image name")
+    parser.add_argument("--openscap-from-git", action="store_true", default=False,
+                        help="Use OpenSCAP from upstream instead of package")
+    parser.add_argument("--ssg-from-git", action="store_true", default=False,
+                        help="Use SCAP Security Guide from upstream instead of package")
+    parser.add_argument("--daemon-from-git", action="store_true", default=False,
+                        help="Use OpenSCAP Daemon from upstream instead of package")
+    args = parser.parse_args()
+
+    f = open("Dockerfile", "w")
+
+    install_command = install_commands.get(args.base, install_commands["rhel"])
+    builddep_package = builddep_packages.get(args.base, builddep_packages["rhel"])
+    builddep_command = builddep_commands.get(args.base, builddep_commands["rhel"])
```

Comment:
I don't think so, because only Fedora has changed the package manager to dnf, but other systems use yum, and yum was here before, so eg. CentOS falls into the default value.